docs/create_your_own_extension_layer.md

```diff
@@ -0,0 +1,177 @@
+# How to build new layer
+
+To create your own extension layer, compile the extension and any required libraries on Bref environment with Docker.
+It picks up pgsql extension as an example, so please replace the name appropriately.
+
+First create new extension directory in `layers/` and move there.
+
+```bash
+$ mkdir layers/pgsql && cd $_
+```
+
+Create `Dockerfile` such as follows.
+
+```Dockerfile
+ARG PHP_VERSION
+FROM bref/build-php-$PHP_VERSION AS ext
+
+#
+# Add commands to build PHP extensions here.
+#
+
+# Build the final image from the lambci image that is close to the production environment
+FROM lambci/lambda:provided
+
+#
+# Add commands to copy files that requried for final image.
+#
+```
+
+Then you need to add two part of Dockerfile, the following snippets are examples of build and file copy parts respectively.
+
+```Dockerfile
+
+WORKDIR ${PHP_BUILD_DIR}/ext/pgsql
+RUN phpize
+RUN ./configure --with-pgsql=${INSTALL_DIR}
+RUN make -j `nproc` && make install
+
+RUN cp `php-config --extension-dir`/pgsql.so /tmp/pgsql.so
+```
+Some env variables are set and can be used in this environment. `PHP_BUILD_DIR` is `/tmp/build/php`, `INSTALL_DIR` is `/opt/bref`.
```

Comment:
We should mention that we get the environment variables from `bref/build-php-$PHP_VERSION`. That will be great if you need to find out what more is available.